installer/generate_deb_control.py

```diff
@@ -0,0 +1,97 @@
+#! /usr/bin/env python3
+
+import datetime
+import platform
+import sys
+import os
+from parse_package_xml import packagexml_parser
+from parse_install_yaml import installyaml_parser
+
+
+TUE_ENV_TARGETS_DIR = os.environ["TUE_ENV_TARGETS_DIR"]
+TUE_ROS_DISTRO = os.environ["TUE_ROS_DISTRO"]
+ARCHITECTURES = {"x86_64": "amd64", "aarch64": "arm64"}
+CONTROL_FILE_TEMPLATE = """
+Package: {package}
+Version: {version}
+Architecture: {arch}
+Maintainer: {maintainer}
+Description: {description}
+Depends: {dependencies}
+"""
+
+
+def main():
+    if len(sys.argv) != 4:
+        print("Usage: generate_deb_control RELEASE_DIR PACKAGE.XML TIMESTAMP")
+        return 1
+
+    # print(generate_control_file(sys.argv[2])["control"])
+    create_dirs(sys.argv[1], sys.argv[2], sys.argv[3])
+
+
+def generate_control_file(path: str) -> dict:
+    parsed = packagexml_parser(path)["parser"]
+
+    maintainer_string = ", ".join(
+        "{maintainer} <{email}>".format(maintainer=m, email=parsed["emails"][m]) for m in parsed["maintainer"]
+    )
+
+    system_deps = []
+    for dep in parsed["depend"]:
+        install_yaml = os.path.join(TUE_ENV_TARGETS_DIR, dep, "install.yaml")
+        if not os.path.exists(install_yaml):
+            install_yaml = os.path.join(TUE_ENV_TARGETS_DIR, "ros-" + dep, "install.yaml")
+        system_deps += installyaml_parser(install_yaml)["system_packages"]
+
+    system_dep_string = ""
+    if system_deps:
+        system_dep_string = ", ".join(set(system_deps))
+
+    version = list(parsed["version"])[0]
+    description = list(parsed["description"])[0]
```

Comment:
Is this robust again IndexError?

Reply:
It is what I hope but will check it again